interpreter/cling/lib/Interpreter/ValuePrinter.cpp

```diff
@@ -344,8 +344,452 @@ static void StreamValue(llvm::raw_ostream& o, const void* V,
+              /*omitDS*/false,
+                                              &Interp.getSema()))) {
+    if (const clang::FunctionDecl* FDsetValue
+        = llvm::dyn_cast_or_null<clang::FunctionDecl>(CallE->getCalleeDecl())){
+      if (FDsetValue->getNameAsString() == "setValueNoAlloc" &&
+          CallE->getNumArgs() == 5) {
+        const clang::Expr* Arg4 = CallE->getArg(4);
+        while (const clang::CastExpr* CastE
+            = clang::dyn_cast<clang::CastExpr>(Arg4))
+          Arg4 = CastE->getSubExpr();
+        if (const clang::DeclRefExpr* DeclRefExp
+            = llvm::dyn_cast<clang::DeclRefExpr>(Arg4))
+          FD = llvm::dyn_cast<clang::FunctionDecl>(DeclRefExp->getDecl());
+      }
+    }
+  }
+
+  if (FD) {
+    clang::SourceRange SRange = FD->getSourceRange();
+    const char* cBegin = 0;
+    const char* cEnd = 0;
+    bool Invalid;
+    if (SRange.isValid()) {
+      clang::SourceManager& SM = C.getSourceManager();
+      clang::SourceLocation LocBegin = SRange.getBegin();
+      LocBegin = SM.getExpansionRange(LocBegin).first;
+      o << "  at " << SM.getFilename(LocBegin);
+      unsigned LineNo = SM.getSpellingLineNumber(LocBegin, &Invalid);
+      if (!Invalid)
+        o << ':' << LineNo;
+      o << ":\n";
+      bool Invalid = false;
+      cBegin = SM.getCharacterData(LocBegin, &Invalid);
+      if (!Invalid) {
+        clang::SourceLocation LocEnd = SRange.getEnd();
+        LocEnd = SM.getExpansionRange(LocEnd).second;
+        cEnd = SM.getCharacterData(LocEnd, &Invalid);
+        if (Invalid)
+          cBegin = 0;
+      } else {
+        cBegin = 0;
+      }
+    }
+    if (cBegin && cEnd && cEnd > cBegin && cEnd - cBegin < 16 * 1024) {
+      o << llvm::StringRef(cBegin, cEnd - cBegin + 1);
+    } else {
+      const clang::FunctionDecl* FDef;
+      if (FD->hasBody(FDef))
+        FD = FDef;
+      FD->print(o);
+      //const clang::FunctionDecl* FD
+      //  = llvm::cast<const clang::FunctionType>(Ty)->getDecl();
+    }
+  }
+  // type-based print() never and decl-based print() sometimes does not include
+  // a final newline:
+  o << '\n';
+  functionString = o.str();
+  return functionString;
+}
+
+static std::string invokePrintValueOverload(const Value& V) {
+  Interpreter *Interp = V.getInterpreter();
+  std::stringstream printValueSS;
+  printValueSS << "cling::printValue(";
+  printValueSS << getCastValueString(V);
+  printValueSS << ");";
+  Value printValueV;
+  Interp->evaluate(printValueSS.str(), printValueV);
+  assert(printValueV.isValid() && "Must return valid value.");
+  return *(std::string*)printValueV.getPtr();
+}
+
 namespace cling {
+
+  std::string printValue(void* ptr) {
+    std::ostringstream strm;
+    if (!ptr) {
+      strm << "<<<NULL>>>";
+    } else {
+      strm << ptr;
+    }
+    return strm.str();
+  }
+
+  // Bool
+  std::string printValue(bool val) {
+    std::ostringstream strm;
+    strm << std::boolalpha;
+    strm << val;
+    strm << std::noboolalpha;
```

Comment:
This is problematic because we don't know whether the user has set `std::boolalpha` themselves, and you are setting it to `std::noboolalpha` unconditionally. Can we just return `"true"` or `"false"`?